### ISSUE TITLE
Explain analyze coredump while direct dispatch is enabled

### ISSUE THREAD

**Issue by hidva** [greenplum-db/gpdb#11460]
### Greenplum version or build

```
tmp=# select version();
PostgreSQL 9.4.24 (Greenplum Database 6.13.0+dev.61.g1c2dc45150 build dev) on x86_64-unknown-linux-gnu, compiled by gcc (GCC) 6.5.1 20190307 (Alibaba 6.5.1-1 2.17), 64-bit compiled on Feb  2 2021 21:07:32 (with assert checking)
```

### OS version and uname -a

```
$ uname  -a
Linux gp6-devecs 4.19.91-007.ali4000.alios7.x86_64 #1 SMP Wed Apr 8 16:17:43 CST 2020 x86_64 x86_64 x86_64 GNU/Linux
```

### autoconf options used ( config.status --config )

```
$ ./config.status --config
'--prefix=/u01/user/zhanyi/dbbin/db' '--enable-gpfdist' '--disable-pxf' '--enable-debug' '--enable-depend' '--without-zstd' '--enable-cassert' '--enable-debug-extensions' '--disable-gpcloud' 'CFLAGS=-O0 -ggdb -fsanitize=address' 'CXXFLAGS=-O0 -ggdb -fsanitize=address' 'LIBS=-ldl'
```

### Installation information ( pg_config )

```
$ pg_config
BINDIR = /u01/user/zhanyi/dbbin/db/bin
DOCDIR = /u01/user/zhanyi/dbbin/db/share/doc/postgresql
HTMLDIR = /u01/user/zhanyi/dbbin/db/share/doc/postgresql
INCLUDEDIR = /u01/user/zhanyi/dbbin/db/include
PKGINCLUDEDIR = /u01/user/zhanyi/dbbin/db/include/postgresql
INCLUDEDIR-SERVER = /u01/user/zhanyi/dbbin/db/include/postgresql/server
LIBDIR = /u01/user/zhanyi/dbbin/db/lib
PKGLIBDIR = /u01/user/zhanyi/dbbin/db/lib/postgresql
LOCALEDIR = /u01/user/zhanyi/dbbin/db/share/locale
MANDIR = /u01/user/zhanyi/dbbin/db/share/man
SHAREDIR = /u01/user/zhanyi/dbbin/db/share/postgresql
SYSCONFDIR = /u01/user/zhanyi/dbbin/db/etc/postgresql
PGXS = /u01/user/zhanyi/dbbin/db/lib/postgresql/pgxs/src/makefiles/pgxs.mk
CONFIGURE = '--prefix=/u01/user/zhanyi/dbbin/db' '--enable-gpfdist' '--disable-pxf' '--enable-debug' '--enable-depend' '--without-zstd' '--enable-cassert' '--enable-debug-extensions' '--disable-gpcloud' 'CFLAGS=-O0 -ggdb -fsanitize=address' 'CXXFLAGS=-O0 -ggdb -fsanitize=address' 'LIBS=-ldl'
CC = gcc
CPPFLAGS = -D_GNU_SOURCE -I/u01/user/zhanyi/dbbin/db/include
CFLAGS = -Wall -Wmissing-prototypes -Wpointer-arith -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -fno-aggressive-loop-optimizations -Wno-unused-but-set-variable -Wno-address -g -ggdb -O0 -ggdb -fsanitize=address -std=gnu99 -Werror=uninitialized -Werror=implicit-function-declaration -I/u01/user/zhanyi/dbbin/db/include
CFLAGS_SL = -fPIC
LDFLAGS = -Wl,--as-needed -Wl,-rpath,'/u01/user/zhanyi/dbbin/db/lib',--enable-new-dtags -L/u01/user/zhanyi/dbbin/db/lib
LDFLAGS_EX =
LDFLAGS_SL =
LIBS = -lpgcommon -lpgport -lnuma -lxerces-c -lrt -lrt -lcrypt -lm -ldl -L/u01/user/zhanyi/dbbin/db/lib
VERSION = PostgreSQL 9.4.24
```

### Actual behavior

```sql
tmp=# create table t( i int , j int ) ;
tmp=# insert into t select i, i from generate_series(1, 1000) i;
tmp=# set optimizer to off;
tmp=# explain select * from t where i = 2 or i = 5;
Gather Motion 2:1  (slice1; segments: 2)  (cost=0.00..18.00 rows=2 width=8)
  ->  Seq Scan on t  (cost=0.00..18.00 rows=1 width=8)
        Filter: ((i = 2) OR (i = 5))
Optimizer: Postgres query optimizer
tmp=# set explain_memory_verbosity to DETAIL;
tmp=# explain analyze select * from t where i = 2 or i = 5;
FATAL:  Unexpected internal error (memaccounting.c:842)
DETAIL:  FailedAssertion("!(longLivingAccount->parentId <= MEMORY_OWNER_TYPE_END_LONG_LIVING)", File: "memaccounting.c", Line: 842)
HINT:  Process 27786 will wait for gp_debug_linger=120 seconds before termination.
Note that its locks and other resources will not be released until then.
server closed the connection unexpectedly
	This probably means the server terminated abnormally
	before or while processing the request.
The connection to the server was lost. Attempting reset: Succeeded.
```

```
z-libs-5.1.2-12alpha.1.alios7.x86_64 zlib-1.2.7-16.2.alios7.x86_64
(gdb) bt
#0  0x00007f367ace2fe3 in select () from /lib64/libc.so.6
#1  0x00000000020e103d in pg_usleep (microsec=5000000) at pgsleep.c:56
#2  0x00000000017b48c8 in elog_debug_linger (edata=0x38247a0 <errordata>) at elog.c:5261
#3  0x000000000179feba in errfinish (dummy=0) at elog.c:670
#4  0x000000000179c5ec in ExceptionalCondition (conditionName=0x24e7020 "!(longLivingAccount->parentId <= MEMORY_OWNER_TYPE_END_LONG_LIVING)", errorType=0x24e5e00 "FailedAssertion", fileName=0x24e5f80 "memaccounting.c", lineNumber=842) at assert.c:46
#5  0x000000000183cf06 in ConvertMemoryAccountArrayToTree (longLiving=0x625000537c20, shortLiving=0x625000537c58, shortLivingCount=18446744073709551610) at memaccounting.c:842
#6  0x000000000183aef2 in MemoryAccounting_CombinedAccountArrayToExplain (accountArrayBytes=0x0, accountCount=0, explainstate=0x7ffd7461eea0) at memaccounting.c:463
#7  0x0000000000cf3d9f in cdbexplain_showExecStats (planstate=0x625000539fe8, es=0x7ffd7461eea0) at explain_gp.c:1553
#8  0x0000000000d031d5 in ExplainNode (planstate=0x625000539fe8, ancestors=0x0, relationship=0x0, plan_name=0x0, es=0x7ffd7461eea0) at explain.c:2169
#9  0x0000000000cfc5c0 in ExplainPrintPlan (es=0x7ffd7461eea0, queryDesc=0x62500052b668) at explain.c:743
#10 0x0000000000cfb753 in ExplainOnePlan (plannedstmt=0x62500052b4f8, into=0x0, es=0x7ffd7461eea0, queryString=0x62500004b148 "explain analyze select * from t where i = 2 or i = 5;", params=0x0, planduration=0x7ffd7461ed30) at explain.c:616
#11 0x0000000000cfa949 in ExplainOneQuery (query=0x6190000a5828, into=0x0, es=0x7ffd7461eea0, queryString=0x62500004b148 "explain analyze select * from t where i = 2 or i = 5;", params=0x0) at explain.c:423
#12 0x0000000000cf9f6d in ExplainQuery (stmt=0x62500004c3a0, queryString=0x62500004b148 "explain analyze select * from t where i = 2 or i = 5;", params=0x0, dest=0x6190000a5758) at explain.c:256
#13 0x000000000144ee65 in standard_ProcessUtility (parsetree=0x62500004c3a0, queryString=0x62500004b148 "explain analyze select * from t where i = 2 or i = 5;", context=PROCESS_UTILITY_TOPLEVEL, params=0x0, dest=0x6190000a5758, completionTag=0x7ffd7461f1d0 "") at utility.c:795
#14 0x000000000144d958 in ProcessUtility (parsetree=0x62500004c3a0, queryString=0x62500004b148 "explain analyze select * from t where i = 2 or i = 5;", context=PROCESS_UTILITY_TOPLEVEL, params=0x0, dest=0x6190000a5758, completionTag=0x7ffd7461f1d0 "") at utility.c:374
#15 0x000000000144b2eb in PortalRunUtility (portal=0x625000070948, utilityStmt=0x62500004c3a0, isTopLevel=1 '\001', dest=0x6190000a5758, completionTag=0x7ffd7461f1d0 "") at pquery.c:1381
#16 0x000000000144ada3 in FillPortalStore (portal=0x625000070948, isTopLevel=1 '\001') at pquery.c:1251
#17 0x0000000001449d04 in PortalRun (portal=0x625000070948, count=9223372036854775807, isTopLevel=1 '\001', dest=0x62500004c890, altdest=0x62500004c890, completionTag=0x7ffd7461f590 "") at pquery.c:987
#18 0x0000000001438bf1 in exec_simple_query (query_string=0x62500004b148 "explain analyze select * from t where i = 2 or i = 5;") at postgres.c:1820
#19 0x00000000014432bc in PostgresMain (argc=1, argv=0x625000023700, dbname=0x625000023590 "tmp", username=0x625000023570 "zhanyi") at postgres.c:5242
#20 0x00000000012ccb68 in BackendRun (port=0x61400000d040) at postmaster.c:4811
#21 0x00000000012cb73c in BackendStartup (port=0x61400000d040) at postmaster.c:4468
#22 0x00000000012c4084 in ServerLoop () at postmaster.c:1948
#23 0x00000000012c2ae9 in PostmasterMain (argc=6, argv=0x60600000eae0) at postmaster.c:1518
#24 0x0000000000fe483b in main (argc=6, argv=0x60600000eae0) at main.c:245
```

### Root Cause

```c
static void
cdbexplain_showExecStats(struct PlanState *planstate, ExplainState *es)
{
        struct CdbExplain_ShowStatCtx *ctx = es->showstatctx;
        Instrumentation *instr = planstate->instrument;
        CdbExplain_NodeSummary *ns = instr->cdbNodeSummary;
        instr_time      timediff;
        int                     i;

        char            totalbuf[50];
        char            avgbuf[50];
        char            maxbuf[50];
        char            segbuf[50];
        char            startbuf[50];

        /* Might not have received stats from qExecs if they hit errors. */
        if (!ns)
                return;

        Assert(instr != NULL);

        if ((EXPLAIN_MEMORY_VERBOSITY_DETAIL <= explain_memory_verbosity)
                && planstate->type == T_MotionState)
        {
                Motion     *pMotion = (Motion *) planstate->plan;
                int                     curSliceId = pMotion->motionID;

                for (int iWorker = 0; iWorker < ctx->slices[curSliceId].nworker; iWorker++)
                {
                        if (es->format == EXPLAIN_FORMAT_TEXT)
                        {
                                appendStringInfoSpaces(es->str, es->indent * 2);
                                appendStringInfo(es->str, "slice %d, seg %d\n", curSliceId, iWorker);
                        }
                        else
                        {
                                ExplainOpenGroup("MemoryAccounting", NULL, false, es);
                                ExplainPropertyInteger("Slice", curSliceId, es);
                                ExplainPropertyInteger("Segment", iWorker, es);
                        }

                        MemoryAccounting_CombinedAccountArrayToExplain(ctx->slices[curSliceId].memoryAccounts[iWorker],
                                                                                                                   ctx->slices[curSliceId].memoryAccountCount[iWorker],
                                                                                                                   es);
                        if (es->format != EXPLAIN_FORMAT_TEXT)
                                ExplainCloseGroup("MemoryAccounting", NULL, false, es);
                }
        }
```

This will coredump because `ctx->slices[curSliceId].memoryAccounts[iWorker = 1]` is uninitialized.

**Comment by JunfengYang** [comment]
Nice cache! Would you like to open PR for the fix? Or we can handle it.

**Comment by hidva** [comment]
Ok, I will open a PR to fix this.

**Comment by JunfengYang** [comment]
Merge the fix as commit 7427fed96c8f69fa77f638d23b5b3f6ebc28cc6b.
